lib/CodeTransform/Adapter/WorseReflection/Refactor/WorseHereDoc.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Phpactor\CodeTransform\Adapter\WorseReflection\Refactor;
+
+use Microsoft\PhpParser\Parser;
+use Microsoft\PhpParser\Node\StringLiteral;
+use Phpactor\CodeTransform\Domain\Refactor\ByteOffsetRefactor;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\TextDocument;
+use Phpactor\TextDocument\TextEdit;
+use Phpactor\TextDocument\TextEdits;
+use Microsoft\PhpParser\Token;
+use Microsoft\PhpParser\MissingToken;
+use Microsoft\PhpParser\TokenKind;
+
+class WorseHereDoc implements ByteOffsetRefactor
+{
+    public function __construct(private Parser $parser)
+    {
+    }
+
+    public function refactor(TextDocument $document, ByteOffset $offset): TextEdits
+    {
+        $node = $this->parser
+            ->parseSourceFile($document->__toString())
+            ->getDescendantNodeAtPosition($offset->toInt())
+        ;
+
+        // If we're inside a variable inside a string literal, get the surrounding string
+        if (!$node instanceof StringLiteral) {
+            $node = $node->getFirstAncestor(StringLiteral::class);
+        }
+
+        if (!$node instanceof StringLiteral) {
+            return TextEdits::none();
+        }
+
+        if ($node->startQuote instanceof Token && $node->startQuote->kind === TokenKind::HeredocStart) {
+            if ($node->endQuote instanceof MissingToken) {
+                return TextEdits::none();
+            }
+            return $this->convertFromHereDocToString($node);
+        }
+
+        return $this->convertFromStringToHereDoc($node);
+    }
+
+    private function convertFromStringToHereDoc(StringLiteral $node): TextEdits
+    {
+        // Trimming the quotes
+        $content = $node->getStringContentsText();
+
+        return TextEdits::fromTextEdits([TextEdit::create(
+            $node->getStartPosition(),
+            $node->getEndPosition() - $node->getStartPosition(),
+            '<<<TEXT'.PHP_EOL.$content.PHP_EOL.'TEXT'
```

Comment:
maybe `EOF` is a more common delimiter. Incidentally Neovim/treesitter uses the delimiter for syntax highlighting.